### ISSUE TITLE
loss这里设计的是不是有问题？

### ISSUE THREAD

**Issue by AiAiGhost** [polarisZhao/PFLD-pytorch#59]
weight_attribute = torch.sum(attributes_w_n.mul(mat_ratio), axis=1)
loss里乘这一项导致没有特殊属性的人脸loss值为0，就是正常的人脸在训练里面的贡献为0，这样的设计正确么？还是我理解有误？

**Comment by Rayson2020-8** [comment]
我也发现了这个问题，很奇怪

**Comment by Rayson2020-8** [comment]
![image](https://user-images.githubusercontent.com/69963621/182827257-93c915e7-d06b-4e83-be51-7fd4ff9c94b4.png)
这个项目作者复现错了，去看了tensorflow的项目，那里最后是使用weighted_loss + l2 loss = total loss进行backward，而不是单单使用weighted_loss,两者加起来才合理。

**Comment by qmcreeper** [comment]
没记错的话torch.cos()期望输入是弧度，数据貌似是角度，直接输入本身有点问题。另外预测值和真实值理论范围是(-inf,+inf)，再加上余弦函数是周期函数，也就是说只要误差接近(1+T)pi损失值weighted_loss会非常接近0

**Comment by Rayson2020-8** [comment]
您好，我已经收到您的来信，稍后会回复您，感谢您的来信
 
施荣盛华南理工大学材料科学与工程学院Tel：
